xbrowse_server/phenotips/utilities.py

```diff
@@ -0,0 +1,121 @@
+      collaborator_username,collab_pwd=get_generic_collaborator_uname_pwd_for_project(project_id)
+      add_read_only_collaborator_phenotips_patient(collaborator_username,patient_eid)
+  else:
+      print 'error creating patient',individual_id,':',result
+  
+      
+#authenticates to phenotips, fetches (GET) given results and returns that
+def do_authenticated_call_to_phenotips(url,uname,pwd):
+  '''authenticates to phenotips, fetches (GET) given results and returns that'''
+  try:
+    password_mgr = urllib2.HTTPPasswordMgrWithDefaultRealm()
+    request = urllib2.Request(url)
+    base64string = base64.encodestring('%s:%s' % (uname, pwd)).replace('\n', '')
+    request.add_header("Authorization", "Basic %s" % base64string)   
+    result = urllib2.urlopen(request)   
+    return result
+  except Exception as e:
+    return e
+    
+
+#to help process a translation of internal id to external id
+def convert_internal_id_to_external_id(int_id,project_phenotips_uname,project_phenotips_pwd):
+  '''to help process a translation of internal id to external id '''
+  try:
+    url= os.path.join(settings.PHENOPTIPS_HOST_NAME,'rest/patients/eid/'+str(int_id))   
+    result = do_authenticated_call_to_phenotips(url,project_phenotips_uname,project_phenotips_pwd)
+    as_json = json.loads(result.read())
+    return as_json['id']
+  except Exception as e:
+    print 'convert internal id error:',e,result
+    logger.error('phenotips.views:'+str(e) + ' : ' + str(result))
+    return {'mapping':None,'error':str(e)}
+  
+#return the username and password for this project
+def get_uname_pwd_for_project(project_name):
+  uname=project_name
+  pwd=project_name+project_name
+  return uname,pwd
+
+
+#return the generic collaborator username and password for this project
+def get_generic_collaborator_uname_pwd_for_project(project_name):
+  uname=project_name+ '_view'
+  pwd=project_name+project_name
+  return uname,pwd
+
+
+
+#generates a new user in phenotips
+def add_new_user_to_phenotips(new_user_first_name, new_user_last_name,new_user_name,email_address,new_user_pwd):
+  #we need to put this password in a non-checkin file
+  admin_uname='Admin'
+  admin_pwd='admin'
+  headers={"Content-Type": "application/x-www-form-urlencoded"}
+  data={'parent':'XWiki.XWikiUsers'}
+  url = settings.PHENOPTIPS_HOST_NAME + '/rest/wikis/xwiki/spaces/XWiki/pages/' + new_user_name
+  do_authenticated_PUT(admin_uname,admin_pwd,url,data,headers) 
+  data={'className':'XWiki.XWikiUsers',
+        'property#first_name':new_user_first_name,
+        'property#last_name':new_user_last_name,
+        'property#email':email_address,
+        'property#password':new_user_pwd
+        }
+  url=settings.PHENOPTIPS_HOST_NAME + '/rest/wikis/xwiki/spaces/XWiki/pages/' + new_user_name + '/objects'
+  do_authenticated_POST(admin_uname,admin_pwd,url,data,headers)
+  
+  
+#adds a collaborator to an existing patient. Requires an existing collaborator username, patient_eid (PXXXX..)
```

Comment:
Just curious - why are all collaborators added as 'read_only' ? 
Also, I really like these method docs, I would just put them in a method doc string.